tests/x11/reboot_gnome.pm

```diff
@@ -14,19 +14,40 @@ use testapi;
 use utils;
 
 sub run() {
-    wait_idle;
-    send_key "ctrl-alt-delete";    # reboot
-    assert_screen 'logoutdialog', 15;
+    wait_still_screen;
+    send_key_until_needlematch 'logoutdialog', 'ctrl-alt-delete', 7, 10;    # reboot
     assert_and_click 'logoutdialog-reboot-highlighted';
 
     if (get_var("SHUTDOWN_NEEDS_AUTH")) {
-        assert_screen 'reboot-auth', 15;
-        sleep 3;
+        assert_screen 'reboot-auth';
+        wait_still_screen;
         type_password;
-        sleep 3;
-        assert_and_click 'reboot-auth-typed', 'right';    # Extra assert_and_click (with right click) to check the correct number of characters is typed and open up the 'show text' option
-        assert_and_click 'reboot-auth-showtext';          # Click the 'Show Text' Option to enable the display of the typed text
-        assert_screen 'reboot-auth-correct-password';     # Check the password is correct
+        wait_still_screen;
```

Comment:
everything above this line looks good to me, about the change afterwards I am not so sure as even if the password typing fails like 5-6 times, who will notice the diag messages anyway?

I suggest to just start with replacing the sleep and ", 15" and keep the later parts only with record soft failures to open bugs before retrying multiple times as workaround. There should be open bugs for at least some of these issues.